### PR TITLE
docs(adr): ADR-009: Supply module

### PR DESCRIPTION
**Pull request by bragaz**
## Description

Closes: #XXXX

ADR-009 for supply module implementation.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/architecture/adr-009-supply-module.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+- March 21, 2022: Initial draft.
+
+## Status
+
+DRAFTED
+
+## Abstract
+
+This ADR defines the `x/supply` module which will expose a set of APIs that will be called by data aggregator websites (such as Coingecko and CoinMarketCap)
+in order to fetch updated data about $DSM supply.
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:14]:
```suggestion
This ADR defines the `x/supply` module which will expose a set of APIs that can be called by data aggregator websites (such as Coingecko and CoinMarketCap) in order to fetch updated data about a specific token supply.
```

File: docs/architecture/adr-009-supply-module.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Context
+
+Currently, inside our [CoinGecko](https://www.coingecko.com/en/coins/desmos) and [CoinMarketCap](https://coinmarketcap.com/currencies/desmos/) 
+some important information (current and total supply) are missing or not correctly updated. To solve this, we can implement a
+series of APIs that fetch those data and can be later used by all the data aggregator websites that has the $DSM token listed.
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:20]:
```suggestion
Currently, inside our [CoinGecko](https://www.coingecko.com/en/coins/desmos) and [CoinMarketCap](https://coinmarketcap.com/currencies/desmos/) pages some important information about current and total supply are missing or not correctly updated. To solve this, we can implement a series of APIs that read those data from the chain and can be later used by all the data aggregator websites that have the token listed.
```

File: docs/architecture/adr-009-supply-module.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Decision
+
+The APIs will be exposed in a new module called `x/supply` that will have the purpose to fetch the given coin info from different 
+cosmos-SDK modules (`x/bank`, `x/distribution`, `x/staking` ) and apply some conversions on them in order to avoid displaying them
+in millionth units.
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:26]:
```suggestion
The APIs will be exposed in a new module called `x/supply` that will have the purpose to fetch the given information using different Cosmos SDK modules (namely `x/bank`, `x/distribution` and `x/staking` ) and apply some conversions on them in order to display them the best way possible for the client.
```

File: docs/architecture/adr-009-supply-module.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+### Queries
+
+#### Total Supply
+This query will:
+1. Fetch the total supply of the given token `denom` from the `bank` module
+2. Convert it in order to display its non-millionth value
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:33]:
```suggestion
This query will fetch the total supply of a given token `denom`.
```

File: docs/architecture/adr-009-supply-module.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+#### Current supply
+This query will:
+1. Fetch the total supply of the given token `denom`
+2. Fetch the total vested tokens amount
+3. Fetch the community pool amount
+4. Calculate the circulating supply by subtracting data of 2. and 3. from Total supply
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:40]:
```suggestion
This query will return the circulating supply amount by subtracting the total vested amount and the community pool amount from the total supply of the given token.
```

File: docs/architecture/adr-009-supply-module.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+## Test Cases
+
+- Check API to correctly return the converted Total Supply from millionth to non-millionth representation;
+- Check API to correctly return the current supply.
+
+
+## References
+
+Actual alpha branch of `x/supply`:  
+https://github.com/desmos-labs/desmos/tree/leonardo/coingecko-APIs/x/supply.
```
Comment (RiccardoM) [inline docs/architecture/adr-009-supply-module.md:57]: It's better to put the issue containing the discussion here
```suggestion
- [Issue #733](https://github.com/desmos-labs/desmos/issues/773)
```

**Review by RiccardoM — APPROVED** [review]

**Review by dadamu — APPROVED** [review]

**Pull request merged**